src/cache.rs

```diff
@@ -0,0 +1,28 @@
+use jsonrpsee::core::JsonValue;
+use lru::LruCache;
+use std::num::NonZeroUsize;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+pub struct Cache {
+    map: Arc<Mutex<LruCache<String, JsonValue>>>,
```

Comment:
use RwLock

Reply:
don't see any benefit to using RwLock since `get` will update position so it required `write`

Reply:
what do you mean? a cache hit will not need write

Reply:
Lru cache removes the least used value so it updates on get

Reply:
I see. In that case we need something that doesn't do that. It will completely kill the performance.

Reply:
We can use peek which doesn't update anything but will lose the feature